### DISCUSSION TITLE
Some misc questions

### DISCUSSION THREAD

**Discussion by Gigas002** [woelper/oculante#248] [Q&A]
Just found out `oculante` and it's amazing, loved it the moment seen it!
I've used `qimgv` for a long time on a `x11`-session, but since I've moved to `wayland`, I needed a different solution. For some time I've used `imv`, but it's a bit abandoned (as a `qimgv`, in fact) and not quite fit my needs. `oculante` is really fast and handles my use-cases very good, but I still have some questions to ask.

First, is there a way to control a zoom-in/out step? It's a bit too "fast" for me.

For next questions, it'd be better to refer to a screenshot below.

![2023-12-01_23-59](https://github.com/woelper/oculante/assets/24297712/e308a998-592b-4784-bc0e-23f82e870e53)

On this screenshot, the **left** one is `oculante`, on the **middle** is `qimgv` and on the **right** is `imv`.

I don't know, how `imv` handles the zoom process, but `qimgv` has nice settings for upscaling process, making the zoomed image prettier. Is there any plans for such feature in `oculante`?

Finally, I wonder, where the border-line between `oculante` and `qimgv` came from. I've disabled window borders with a hotkey, so there shouldn't be any borders, like between middle and right picture, but it exists. Not sure, if it's a bug, but would be better, if it wouldn't exist.

**Comment by Gigas002** [comment]
After using the app for some time, found out some more things to ask/suggest. Yet again, I'm not complaining, I don't think there are a lot of people in need of these features except for me. Just wondering

1. Is there a specific reason for setting minimal/maximal window sizes? I've seen other image viewers do so - actually, only qimgv could handle resizing the window to a minimal value, around 1x1 pixel I guess. Would be great if there wouldn't be a minimal window size limit
2. The zoomed-in picture gets to it's default state, as window resizes. While I think it's suitable for most people, sometimes you want zoomed picture to stay in that state, while resizing the window. Maybe a setting for controlling such behavior would be enough? UPD. Found out that this happens only in zen-mode, the default view with panel doesn't change the image sizes on window resize
3. Any plans for supporting the [DNG](https://en.wikipedia.org/wiki/Digital_Negative) RAW format?

**Comment by woelper** [comment]
Hi @Gigas002 ! Thank you for taking the time to write here and for the kind words! Some great points here, and I'll try answering them together!

> First, is there a way to control a zoom-in/out step? It's a bit too "fast" for me.

No. To be fair, I had some trouble with zooming, and I am not yet happy. For example on Mac,the zoom is still faster for the image, while scrolling in dropdowns is too slow. Having a zoom multiplier in the options could be a solution maybe?

> I don't know, how imv handles the zoom process, but qimgv has nice settings for upscaling process, making the zoomed image prettier. Is there any plans for such feature in oculante?

This is by design. I've initially written oculante for me to inspect images (I make games) and I wanted to precisely distinguish between pixels. I realize this does not work for everyone. I can add an option to control  this behavior.

> Finally, I wonder, where the border-line between oculante and qimgv came from

To be honest, I have no idea. I've never used it borderless. I am working on a proper "no-window-decoration" mode where I internally disable all windowing. For that, I'd need to allow dragging and closing from the menu bar, and I am not happy with it still.

> Is there a specific reason for setting minimal/maximal window sizes? 

No. Not that I know. I've never thought about that, that's why I like these discussions. Maybe these are limits made by the windowing library I use, but I will check.

> The zoomed-in picture gets to it's default state, as window resizes. 

Exactly as you said, only in zen mode. I thought this would be a better default. Should I revert that and not resize?

> Any plans for supporting the [DNG](https://en.wikipedia.org/wiki/Digital_Negative) RAW format?

I am using the `quickraw` library in oculante, which apparently should support dng. I downloaded some example images and they did not work (I get `"Cannot read the raw file."`). Are you seeing similar errors?

**Reply by Gigas002** [reply]
Awesome! Thank you for your detailed response!

> Having a zoom multiplier in the options could be a solution maybe?

Yes, I think that would be the best solution.

> I realize this does not work for everyone. I can add an option to control this behavior

Not sure, how hard it is to implement a proper interpolation algorithm. If it won't be a burden that would be amazing, but don't worry too much about this, I think the current behavior is good enough too.

> Exactly as you said, only in zen mode. I thought this would be a better default. Should I revert that and not resize?

While that indeed fits my needs, I think it's common for someone to see picture auto-resized on window size change. Would be better, if user could control this from settings, I guess.

> I am using the quickraw library in oculante, which apparently should support dng. I downloaded some example images and they did not work (I get "Cannot read the raw file."). Are you seeing similar errors?

Yes. If there's a way to get a more detailed log, I'd gladly provide it.

**Reply by woelper** [reply]
I changed the following:
- Min size is now 1x1, max size is not restricted
- Zen mode does not resize image, instead this is a setting now
- You can adjust the upscaling filter in the settings (Linear/Nearest)
- There is a zoom multiplier

These will be in the next version.

**Reply by Gigas002** [reply]
Awesome! Going to test it today

**Reply by Gigas002** [reply]
> To be honest, I have no idea. I've never used it borderless. I am working on a proper "no-window-decoration" mode where I internally disable all windowing. For that, I'd need to allow dragging and closing from the menu bar, and I am not happy with it still.

Interesting enough, the border is observed only on up and right sides. The left and bottom ones are not affected.

How about configurable hotkeys for that? I don't know if it's possible to do from application though. On my system I use something like this:

`SUPER + H` -- disables window decorations
`SUPER + ]` -- drag the window on the screen
`SUPER + \` -- change window sizes (down-left angle is used)
`Esc` -- close the window

But that's controlled through KDE Plasma hotkeys settings and can be applied to any application.
